### PR TITLE
[CLI] add stop-on-failure flag

### PR DESCRIPTION
**Pull request by r-f-g**

### CODE REVIEW

File: juju_verify/juju_verify.py [hunk 1]
```diff
@@ -121,6 +122,9 @@ def parse_args() -> argparse.Namespace:
     parser.add_argument('-l', '--log-level', type=str.lower,
                         help='Set amount of displayed information',
                         default='info', choices=['trace', 'debug', 'info'])
+    parser.add_argument("--stop-on-failure", action="store_true",
```
Comment (auria) [inline juju_verify/juju_verify.py:125]: Should we come up with a short version of this? `-f` is generally related to `--force` but we could use sit as `fast-fail`. Alternatively, `-s` from `stop` ?

File: juju_verify/juju_verify.py [hunk 2]
```diff
@@ -121,6 +122,9 @@ def parse_args() -> argparse.Namespace:
...
+                        help="Prevent all checks from running and stop at the first"
```
Comment (auria) [inline juju_verify/juju_verify.py:126]: "stop running checks after a failed one" ?

**Review by auria — APPROVED** [review]
Please find some inline comments, which are not a blocker. It would also be nice to add some documentation around the new CLI argument, and how it could be enabled if using the python package.

Other than the above, +1, thank you.

File: juju_verify/juju_verify.py (hunk 1 above)
Comment (r-f-g) [reply juju_verify/juju_verify.py:125]: I don't like using `-f`, because it usually represents `--force` or `--format`. Using `-s` sound good to me.

**Comment by r-f-g** [comment]
> Please find some inline comments, which are not a blocker. It would also be nice to add some documentation around the new CLI argument, and how it could be enabled if using the python package.
> 
> Other than the above, +1, thank you.

After merging #16, I will add documentation. (There's a part where it could land.)

**Review by r-f-g — COMMENTED** [review]
recheck-full

File: docs/getting-started.rst [hunk 3]
```diff
@@ -121,6 +122,34 @@ There are two ways of using them:
   # or
   juju-verify reboot --units ceph-osd/0 --units ceph-osd/1
 
+Stop on failure
+"""""""""""""""
+
+There is an option to stop running checks after a first failed one.
+
+Below you can see an example of use without and with a flag ``--stop-on-failure``.
```
Comment (auria) [inline docs/getting-started.rst:130]: Suggestion: "Find below the difference in behavior when the flag is used."

**Review by auria — APPROVED** [review]
Please find a suggestion to rephrase a sentence in the documentation.

**Review by r-f-g — COMMENTED** [review]
recheck-full

**Review by auria — APPROVED** [review]
+1

**Pull request merged**